File: repos_estudiantes_ALVAREZ_segunda_entrega/CANAVIRI_ALTAMIRANO_AYLIN_MICHEL/COMPOSICION-AGREGACION/EJERCICIO3/JAVA-EJERCICIO3/src/Ejercicio3/Ej3.java
package Ejercicio3;

import java.util.ArrayList;

public class Ej3 {
    public static void main(String[] args) {
        Avion avion1 = new Avion("Boeing 747", "Boeing");
        Parte motor = new Parte("Motor", 4000);
        Parte alas = new Parte("Alas", 15000);
        Parte tren_aterrizaje = new Parte("Tren de Aterrizaje", 3000);

        avion1.agregar_parte(motor);
        avion1.agregar_parte(alas);
        avion1.agregar_parte(tren_aterrizaje);

        avion1.mostrar_avion();
    }
}

class Parte {
    private String nombre;
    private int peso;

    public Parte(String nombre, int peso) {
        this.nombre = nombre;
        this.peso = peso;
    }

    public String get_nombre() {
        return nombre;
    }

    public void set_nombre(String nombre) {
        this.nombre = nombre;
    }

    public int get_peso() {
        return peso;
    }

    public void set_peso(int peso) {
        this.peso = peso;
    }

    public void mostrar_info() {
        System.out.println("Parte: " + nombre + ", Peso: " + peso + " kg");
    }
}

class Avion {
    private String modelo;
    private String fabricante;
    private ArrayList<Parte> partes;

    public Avion(String modelo, String fabricante) {
        this.modelo = modelo;
        this.fabricante = fabricante;
        this.partes = new ArrayList<>();
    }

    public String get_modelo() {
        return modelo;
    }

    public void set_modelo(String modelo) {
        this.modelo = modelo;
    }

    public String get_fabricante() {
        return fabricante;
    }

    public void set_fabricante(String fabricante) {
        this.fabricante = fabricante;
    }

    public ArrayList<Parte> get_partes() {
        return partes;
    }

    public void agregar_parte(Parte parte) {
        partes.add(parte);
    }

    public void mostrar_avion() {
        System.out.println("Modelo: " + modelo);
        System.out.println("Fabricante: " + fabricante);
        System.out.println("Partes del avión:");
        for (Parte parte : partes) {
            parte.mostrar_info();
        }
    }
}

File: repos_estudiantes_ALVAREZ_segunda_entrega/CANAVIRI_ALTAMIRANO_AYLIN_MICHEL/COMPOSICION-AGREGACION/EJERCICIO3/Ej3.py
class Parte:
    def __init__(self, nombre, peso):
        self.__nombre = nombre
        self.__peso = peso

    def get_nombre(self):
        return self.__nombre

    def get_peso(self):
        return self.__peso

    def set_nombre(self, nombre):
        self.__nombre = nombre

    def set_peso(self, peso):
        self.__peso = peso

    def mostrar_info(self):
        print(f"Parte: {self.__nombre}, Peso: {self.__peso} kg")


class Avion:
    def __init__(self, modelo, fabricante):
        self.__modelo = modelo
        self.__fabricante = fabricante
        self.__partes = []

    def get_modelo(self):
        return self.__modelo

    def get_fabricante(self):
        return self.__fabricante

    def get_partes(self):
        return self.__partes

    def set_modelo(self, modelo):
        self.__modelo = modelo

    def set_fabricante(self, fabricante):
        self.__fabricante = fabricante

    def agregar_parte(self, parte):
        self.__partes.append(parte)

    def mostrar_avion(self):
        print(f"Modelo: {self.__modelo}")
        print(f"Fabricante: {self.__fabricante}")
        print("Partes del avión:")
        for parte in self.__partes:
            parte.mostrar_info()



avion1 = Avion("Boeing 747", "Boeing")
motor = Parte("Motor", 4000)
alas = Parte("Alas", 15000)
tren_aterrizaje = Parte("Tren de Aterrizaje", 3000)

avion1.agregar_parte(motor)
avion1.agregar_parte(alas)
avion1.agregar_parte(tren_aterrizaje)

avion1.mostrar_avion()